src/shogun/mathematics/Math.h

```diff
@@ -1162,25 +1017,30 @@ class CMath : public CSGObject
 			return 0 == a ? b : a;
 		}
 
-		/** Permute randomly the elements of the vector. If provided, use the
-		 * random object to generate the permutations.
-		 * @param v the vector to permute.
-		 * @param rand random object that might be used to generate the permutations.
-		 */
-		template <class T>
-			static void permute(SGVector<T> v, CRandom* rand=NULL)
+		template <
```

Comment:
Hi @vigsterkr , I somehow miss this comment. Actually this function is a part of math before refactor, I just move its implementation to .h due to we need use template parameter in it. So do we need to move it to CRandom and keep it(the random class) ?